src/DZ_Object/Book.java

```diff
@@ -1,37 +1,38 @@
 package DZ_Object;
 
 public class Book {
-    private String author;
+
+    private Author author;
     private String book;
     private int year;
 
-    public Book(String author, String book, int year){
-        if (author.isEmpty() || book.isEmpty() || year <= 0) {
-            System.out.println("Такую книгу нельзя создать");
-            String[] args = {};
-            DZ_Object.Object.main(args);
-        } else {
-            this.author = author;
-            this.book = book;
-            this.year = year;
-
-        }
+    public Book(Author author, String book, int year) {
+        this.author = author;
+        this.book = book;
+        this.year = year;
 
     }
 
-    public String getBook(){
-        return book;
-    }
+public Author getAuthor() {
+    return author;
+}
 
-    public int getYear() {
-        return year;
-    }
+public String getBook(){
+        return book;
+}
 
-    public int setYear(int year) {
-        this.year = year;
-        return year;
-    }
+public int getYear() {
+    return year;
+}
 
+public int setYear(int year) {
+    this.year = year;
+    return year;
+}
```

Comment:
сеттер неверный, он не должен ничего возвращать. Вот так должно быть
```suggestion
public void setYear(int year) {
    this.year = year;
}
```

Reply:
Точно, забыл исправить. Спасибо.